sendbuffer.go

```diff
@@ -34,7 +33,7 @@ type sendBuffer struct {
 	writeTimer     *time.Timer
 }
 
-func newSendBuffer(defaultHeader []byte, w io.Writer, windowSize int) *sendBuffer {
+func newSendBuffer(defaultHeader []byte, out chan []byte, windowSize int) *sendBuffer {
```

Comment:
sendbuffer didn't have any closing problems, but I figured I'd keep the API consistent by passing in the out channel here rather than an io.Writer.